src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -142,10 +142,14 @@ protected void messageReceivedDecorate(final T request, final TransportRequestHa
             //bypass non-netty requests
             if(channelType.equals("direct")) {
                 final String userHeader = getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_USER_HEADER);
+                final String injectedRolesHeader = getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROLES_HEADER);
                 final String injectedUserHeader = getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER_HEADER);
 
                 if(Strings.isNullOrEmpty(userHeader)) {
-                    if(!Strings.isNullOrEmpty(injectedUserHeader)) {
+                    if(!Strings.isNullOrEmpty(injectedRolesHeader)) {
+                        getThreadContext().putTransient(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROLES, injectedRolesHeader);
+                    }
+                    else if(!Strings.isNullOrEmpty(injectedUserHeader)) {
```

Comment:
When both user and role are injected what is the behaviour ? should we fail ?

Reply:
Currently, user injection is controlled via configuration flag (false by default) and we are taking precedence on role injection.